librz/core/core.c

```diff
@@ -2446,3 +2446,36 @@ RZ_API RzCmdStatus rz_core_core_plugins_print(RzCore *core, RzCmdStateOutput *st
 	rz_cmd_state_output_array_end(state);
 	return RZ_CMD_STATUS_OK;
 }
+
+/**
+ * \brief Filters the given output string based on the provided filter.
+ *
+ * \param output The output string to be filtered.
+ * \param filter The filter string to be used for filtering the output.
+ * \return char* The filtered output string. The caller is responsible for freeing this string.
+ */
+RZ_API char *rz_core_filter_string_output(RZ_NONNULL const char *output, const char *filter) {
```

Comment:
Ownership mark for the return, also `output` is probably not a good name in this case, `str` maybe a better one.